crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -33,14 +33,11 @@ pub(crate) async fn farm(
     let base_directory = crate::utils::get_path(custom_path);
 
     let identity = Identity::open_or_create(&base_directory)?;
-    let address: Sha256Hash = identity
-        .public_key()
-        .as_ref()
-        .to_vec()
-        .try_into()
+    let address = <&[u8; SHA256_HASH_SIZE]>::try_from(identity.public_key().as_ref())
         .expect("Length of public key is always correct");
+    let address = (*address).into();
```

Comment:
This is ugly, just use `.to_bytes()` instead to get an array. Here and in other places too, including comments.